### ISSUE TITLE
Customize Backend Widgets/ Visibility For Clients

### ISSUE THREAD

**Issue by mcl11** [elementor/elementor#7670]
Hi,

I do not know if this had ever been requested  but I got a little stuck in code... ;)

What I intended to achieve was to hide specific widget settings control such as margin, typo size and so on for everyone else like Admins. I would handle that completely over custom css and would insist the default functions to admins.

The intention is to simplify the customization for my clients to a minimum. So It's easier and  much more clearer for them and they do not break something...

I could succeed in hiding whole widgets, even remove custom widget settings control.
But if I remove a specific widget setting control, the setting control won't get rendered in Live-Preview-Window-Section. It behaves like the setting never had exists before. Whereas in the frontend they show up correctly.

What I do is just hooking into `elementor/element/after_section_end` and iterating over `$element -> get_section_controls ( $section_id )` and calling `$element -> remove_control ( $type );`

The widget wettings control hides, but does not get rendered...

I know similar features from visual composer in which I have fully control over visibility depending on special conditions like admins accessibility etc. -  nearly fully customizable.

Maybe it's actually not your intention to provide such a feature but it would be awesome if one could customize the backend capabilities to meet one customers needs...

Maybe there is already a possibility that I just have not found yet ;)

Thanks in advance and kind regards
Michael

**Comment by thimfy** [comment]
Do not purchase elementor or any item from creator of it, no support nothing just kind of shit only.

**Comment by mcl11** [comment]
Thx @thimfy ;)

Well, I found similar entries but no luck in a solved contribution...
From a developer's point of view, mhh, a lot seems to be missing. I am still undecided to use elementor for a current project, as it is not yet clear to simplify it for the customer as far as possible.

Too bad. ;)

Greets
Michael

**Comment by shilo-ey** [comment]
Hi @mcl11 

Thanks for your feedback.

As mentioned in issue #7530 currently there is no option to do so via our editor.

As for your request, you can use this code to hide specific widgets for your users which are not "Administrator" roles:

```PHP
add_filter( 'elementor/editor/localize_settings', function( $settings ) {
    $user = wp_get_current_user();
    if ( ! in_array( 'administrator', $user->roles ) ) {
        //The user is not the "administrator" role
        $settings['widgets']['heading']['show_in_panel'] = false;
        return $settings;
    }
} );
```

Paste this code for example for hiding the heading widget.

Thanks!

**Comment by mcl11** [comment]
Hi THX a lot...

I know hiding specific widgets. Works like a charm.... ;) But is there some functionality to hide specific controls of a widget too? With `$element -> remove_control ( $type )` you can remove a control completely. But If you do so, the attribute doesn't get rendered either. I Think there should be a possibility to update a control/ Section and hide just specific controls without removing them completely - to say just the same as you mentioned above with show_in_panel but for certain controls...

Kind Regards
Michael

**Comment by shilo-ey** [comment]
Currently there's not any recommended way to do that.

We will consider adding the option to do so in our future versions (update the API).

Thanks!

**Comment by mcl11** [comment]
THX for the info... ;)

Could you give an approximately date or timing when this feature is planned to be available? 
Some months, next year ... tomorrow ;)

Kind regards and many thx
Michael

**Comment by black-eye** [comment]
+1

**Comment by SmukMarian** [comment]
> Hi @mcl11
> 
> Thanks for your feedback.
> 
> As mentioned in issue #7530 currently there is no option to do so via our editor.
> 
> As for your request, you can use this code to hide specific widgets for your users which are not "Administrator" roles:
> 
> ```
> add_filter( 'elementor/editor/localize_settings', function( $settings ) {
>     $user = wp_get_current_user();
>     if ( ! in_array( 'administrator', $user->roles ) ) {
>         //The user is not the "administrator" role
>         $settings['widgets']['heading']['show_in_panel'] = false;
>         return $settings;
>     }
> } );
> ```
> 
> Paste this code for example for hiding the heading widget.
> 
> Thanks!

Could you please explain where I need to paste this code?

**Comment by nicholaszein** [comment]
Hi! 🙂 

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor v3.18.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
